week6/woo-suk/news/js/main.js

```diff
@@ -0,0 +1,106 @@
+// NEWS API
+const NEWS_API_KEY = '574be9e95f0845e682cdeba35bcf0eab';
+const NEWS_API_URL = `https://newsapi.org/v2/everything?apiKey=${NEWS_API_KEY}`;
+
+// DOM
+const searchInput = document.querySelector('.search-input');
+const searchBtn = document.querySelector('.search-btn');
+const main = document.querySelector('.main');
+
+// EventListen
+searchInput.addEventListener('keypress', (event) => {
+  if (event.key === 'Enter') newsHandler(event);
+});
+searchBtn.addEventListener('click', (event) => {
+  newsHandler(event);
+});
+
+async function newsHandler(event) {
+  const newsJson = await searchNews();
+  if (newsJson === undefined) return;
+  if (newsJson.totalResults === 0) {
+    renderEmpty();
+  } else {
+    renderNews(newsJson.articles);
+  }
+}
+
+// 뉴스 검색
+async function searchNews() {
+  const today = toStringByFormatting(new Date());
+  const word = searchInput.value.trim();
+
+  if (word.length === 0) {
+    alert('검색어를 입력 후 검색해주세요.');
+    searchInput.value = '';
+    searchInput.focus();
+    return;
+  }
+
+  try {
+    const searchUrl = `${NEWS_API_URL}&q=${word}&from=${today}&to=${today}&sortBy=popularity&language=en`;
+    const res = await fetch(searchUrl);
+    if (res.status !== 200) {
+      throw new Error("Can't find news");
+    }
+    const json = await res.json();
+    return json;
+  } catch (e) {
+    console.error(e);
+  }
+}
+
+// 날짜 포맷 변경
+function toStringByFormatting(date, delimiter = '-') {
+  const year = date.getFullYear();
+  const month = leftPad(date.getMonth() + 1);
+  const day = leftPad(date.getDate());
+  return [year, month, day].join(delimiter);
+}
+
+// 10 미만 일,월 포맷 변경
+function leftPad(value) {
+  return value >= 10 ? value : `0${value}`;
+}
+
+function renderEmpty() {
+  main.innerHTML = `
+    <div class="no-article">
+      Can't find articles corresponding to the search word...
+    </div>
+  `;
+}
+
+function renderNews(newsList) {
+  const news = newsList.reduce((prev, news, idx) => {
+    let articleStartTag = '';
+    if (idx === 0) {
+      articleStartTag = `<a class="article highlight-article" href=${news.url}>`;
+    } else {
+      articleStartTag = `<a class="article" href=${news.url}>`;
+    }
+    return (
+      prev +
+      `
+        ${articleStartTag}
```

Comment:
별도의 변수로 태그의 시작과 끝 부분을 분리하기 보다는, 필요한 부분에 한해 조건문을 적용해보면 어떨까요?

```suggestion
    return (
      prev +
      `
        <a class="article ${idx === 0 ? 'highlight-article' : ''}" href=${news.url}>
```

Reply:
오... 저도 클래스명 하나 때문에 너무 장황하게 작성된게 아닌가 고민할 필요가 있다고 생각하고 있었는데 시원하게 긁어주시네요! 삼항연산자로 변경하니 더 간단하고 가독성이 좋아진것 같습니다. 바쁜 와중에 리뷰해주셔서 감사합니다.

Reply:
빠르게 반영해주셔서 감사합니다🙂